### PR TITLE
fix #650: Softkeyboard blocks the UI

### LINKED ISSUE #650: Softkeyboard stays on screen
**Summary:**

Softkeyboard stays on screen even when new fragment loaded 

**Steps to reproduce:**

1) Open any form in app
2) Open Navigation Fragment 
3) Click on any other option

**Expected behaviour:**

Softkeyboard should close whenever change of fragment 

**Observed behaviour:**

Instead, Softkeyboard stays on the screen

**Device and Android version:**

Xiaomi, Marshmallow

**Screenshots:**

**(Same issue persists in Beneficiaries also)**

keyboard_issue

### PR DESCRIPTION
**Pull request by miPlodder**
Fixes #650 

pr_softkeyboard

pr_softkeyboard_2

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by miPlodder** [comment]
@dilpreet96 @satyan @droidchef Please review this PR.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 1]
```diff
@@ -258,6 +259,9 @@ public void onDrawerClosed(View drawerView) {
             @Override
             public void onDrawerOpened(View drawerView) {
                 super.onDrawerOpened(drawerView);
+                InputMethodManager imm = (InputMethodManager) getSystemService(
+                        Context.INPUT_METHOD_SERVICE);
+                imm.hideSoftInputFromWindow(drawerView.getWindowToken(), 0);
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: Please make method in Base Activity and use it here 
```
public void hideKeyboard(View view) {
        InputMethodManager inputManager = (InputMethodManager) this
                 .getSystemService(Context.INPUT_METHOD_SERVICE);
        inputManager.hideSoftInputFromWindow(view.getWindowToken(), InputMethodManager
                 .RESULT_UNCHANGED_SHOWN);
    }
```

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by miPlodder** [comment]
@therajanmaurya I have made the requested changes. Please review this PR.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 2]
```diff
@@ -258,6 +259,8 @@ public void onDrawerClosed(View drawerView) {
             @Override
             public void onDrawerOpened(View drawerView) {
                 super.onDrawerOpened(drawerView);
+
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: Remove this blank link.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 3]
```diff
@@ -462,4 +465,14 @@ public void onReceive(Context context, Intent intent) {
         }
     };
 
+    public void hideKeyboard(View view) {
+
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: Remove this blank link.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 4]
```diff
@@ -462,4 +465,14 @@ public void onReceive(Context context, Intent intent) {
...
+        InputMethodManager inputManager = (InputMethodManager) this
+                .getSystemService(Context.INPUT_METHOD_SERVICE);
+
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: Remove this blank link.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 5]
```diff
@@ -462,4 +465,14 @@ public void onReceive(Context context, Intent intent) {
...
+        inputManager.hideSoftInputFromWindow(view.getWindowToken(), InputMethodManager
+                .RESULT_UNCHANGED_SHOWN);
+
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: Remove this blank link.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 6]
```diff
@@ -462,4 +465,14 @@ public void onReceive(Context context, Intent intent) {
...
+    }
+
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: Remove this blank link.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by miPlodder** [comment]
@therajanmaurya Please review again. I have made the requested changes.

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**